### PR TITLE
feat(components/atom/videoPlayer): Add support for adaptative video streaming

### PR DESCRIPTION
**Pull request by oegea**
### Description, Motivation and Context

**Context**

Implementing video-related features on Adevinta products.

**Description**

Adding support for adaptative video streaming.

### Types of changes

- [X] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by davidmartin84 — APPROVED** [review]

File: components/atom/videoPlayer/demo/index.js [hunk 1]
```diff
@@ -6,13 +6,16 @@ export default () => {
   const [selectedFile, setSelectedFile] = useState('')
 
   return (
-    <>
+    <div style={{maxWidth: '500px'}}>
```
Comment (oegea) [inline components/atom/videoPlayer/demo/index.js:9]: This is temp.

File: components/atom/videoPlayer/src/components/HLSPlayer.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        hls.destroy()
+      }
+    }
+  }, [autoPlay, hlsConfig, playerRef, src])
+
+  // If Media Source is supported, use HLS.js to play video
+  if (Hls.isSupported())
+    return <video title="HLS video player" ref={playerRef} {...props} />
+
+  // Fallback to using a regular video player if HLS is supported by default in the user's browser
```
Comment (jordevo) [inline components/atom/videoPlayer/src/components/HLSPlayer.js:82]:
```suggestion
  // Fallback to using a regular video player if HLS is not supported by default in the user's browser
```

File: components/atom/videoPlayer/src/components/NativePlayer.js [hunk 3]
```diff
@@ -3,13 +3,19 @@ import {useRef} from 'react'
...
 const NativePlayer = ({src}) => {
   const videoNode = useRef(null)
   const videoSrc = useGetBlobAsVideoSrcEffect({src, videoNode})
 
   return (
-    <video ref={videoNode} title="Native video player" controls>
+    <video
+      className={`${BASE_CLASS}-nativePlayer`}
+      ref={videoNode}
+      title="Native video player"
```
Comment (jordevo) [inline components/atom/videoPlayer/src/components/NativePlayer.js:16]: I wonder if we could keep those `title` values (as in https://github.com/SUI-Components/sui-components/pull/2529/files#diff-f06711f0e52367b36a693d58e6b8fe7ac792d218252b72cb7beaaf28b66174c4R80 ) on some config file?

File: components/atom/videoPlayer/src/components/YouTubePlayer.js [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
-      title="YouTube video player"
-      frameBorder="0"
-      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
-      allowFullScreen
-    />
+    <div className={`${BASE_CLASS}-youtubePlayer`}>
+      <iframe
+        className={`${BASE_CLASS}-youtubePlayerFrame`}
+        src={videoSrc}
+        title="YouTube video player"
```
Comment (jordevo) [inline components/atom/videoPlayer/src/components/YouTubePlayer.js:16]: same as above...

**Review by jordevo — APPROVED** [review]

File: components/atom/videoPlayer/demo/index.js (hunk 1 above)
Comment (andresin87) [inline components/atom/videoPlayer/demo/index.js:9]: https://github.com/SUI-Components/sui-components/blob/master/components/atom/progressBar/demo/index.js#L11

take a look on the demo structure

File: components/atom/videoPlayer/package.json [hunk 5]
```diff
@@ -15,6 +15,9 @@
     "type": "git",
     "url": "https://github.com/SUI-Components/sui-components"
   },
+  "dependencies": {
+    "hls.js": "^0.14.17"
```
Comment (andresin87) [inline components/atom/videoPlayer/package.json:19]: remove the ^

**Pull request merged**